### PR TITLE
Resolve property placeholders in EmbeddedKafka topics

### LINKED ISSUE #555: Resolve property placeholders in EmbeddedKafka topics
I'm working on a project where we are configuring all of our topics in property files, so our tests have to know about the contents of the property files to configure the correct topics. The alternative to this is to use `@TestPropertySource` to override our topic properties, but we have a lot of topics, so it's not that convenient. 

Ideally the topic attribute of `@EmbeddedKafka` would be able to accept and resolve property placeholders.

Example:
```java
@EmbeddedKafka(topics = { "${kafka.topic.one}", "${kafka.topic.two}" })
```

### PR DESCRIPTION
**Pull request by zacholauson**
Fixes #555

* Update EmbeddedKafkaContextCustomizer to resolve property placeholders
* Update KafkaStreamsTests to use topic from property
* Add property resolving details to EmbeddedKafka javadoc and testing.adoc

### CODE REVIEW

File: spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java [hunk 1]
```diff
@@ -85,6 +85,8 @@
 	int partitions() default 2;
 
 	/**
+	 * Topics that should be created
+	 * Topics may contain property placeholders, e.g. {@code topics = "${kafka.topic.one:topicOne}"}
```
Comment (zacholauson) [inline spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java]: I wasn't sure if javadoc counted as a cosmetic change, so I didn't add an `@author` tag. Let me know if that needs to be added.

Comment (artembilan) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java]: That’s change anyway, so, please, add your name to the `@author`. Thanks

Comment (zacholauson) [reply spring-kafka-test/src/main/java/org/springframework/kafka/test/context/EmbeddedKafka.java]: Added. Thanks @artembilan

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/94d4fce8930e5b90a860c87649c7d1634cb98343.

Thank you for contribution, @zacholauson .
Looking forward for more!

**Pull request closed without merging**